bokehjs/test/models/ranges/data_range1d.coffee

```diff
@@ -263,6 +258,40 @@ describe "datarange1d module", ->
       expect(r._compute_plot_bounds([g1], bds)).to.be.deep.equal {minX: 0, maxX: 10, minY: 5, maxY: 6}
       expect(r._compute_plot_bounds([g1, g2], bds)).to.be.deep.equal {minX: 0, maxX: 15, minY: 5, maxY: 6}
 
+  describe "update", ->
+
+    it "should update its start and end values", ->
+      r = new DataRange1d()
+      p = new TestObject()
+      g = new GlyphRenderer()
+      g.id = 1
+      p.renderers = [g]
+      r.plots = [p]
+
+      bds = {
+        1: {minX: -10, maxX: -6, minY: 5, maxY: 6}
+      }
+
+      r.update(bds, 0, 1)
+      expect(r.start).to.be.equal -10.2
+
+    it "should not update its start or end values to NaN when log", ->
+      r = new DataRange1d()
+      r._mapper_type = "log"
+      p = new TestObject()
+      g = new GlyphRenderer()
+      g.id = 1
+      p.renderers = [g]
+      r.plots = [p]
+
+      bds = {
+        1: {minX: -10, maxX: -6, minY: 5, maxY: 6}
+      }
+
+      r.update(bds, 0, 1)
+      expect(r.start).to.be.equal r.start #test will fail when r.start is NaN because NaN != NaN
```

Comment:
Better would be to explicitly test for `NaN` with `expect(r.start).not.to.be.NaN`. The code above seems functionally equivalent, but tells a very different story. This is why you need the comment. Avoid such comments as much as possible and just write code that tell the same story. Just to be clear, I don't discourage you to write comments at all, just this particular class of comments.